### DISCUSSION TITLE
http:// added to emptied Website field

### DISCUSSION THREAD

**Discussion by vonscriptor** [solspace/craft-freeform#1586] [Questions & Answers]
### What happened?

When a Website field doesn't have a value (no longer has a value?) Freeform seems to fill it with http:// which isn't a valid URL.

### How can we reproduce this?

1. Create a Website field.
2. See that http:// is auto-filled (or is when a value is removed and the form is refreshed).
3. The form will not submit due to a non-valid URL.

### Environment

- Freeform Pro 5.6.6
- Craft 4.12.7

**Comment by kjmartens** [comment]
Hi @vonscriptor,

I cannot duplicate this issue at all, and it seems strange that Freeform would behave this way. I'm curious how you have configured the form in the builder and if you have any sort of JS on the front end that might be causing this to happen. 🙂

**Reply by vonscriptor** [reply]
Hi, no JS or anything else that should be affecting the values that I'm aware of. I have the Freeform Website field type which has a User Integration that maps to a Craft URL field type (Allowed URL Types: Web page) that is on User Fields.

When I save the Freeform form on the front-end as a user without a Website field value (I'm not sure if a value had to previously exist) then when the page refreshes its value is now "http://".

I change the User's URL field (that Freeform's maps to) to plain text and the issue doesn't happen. It seems Craft's URL field forces a value of `http://`. Should I be using plain text then?

**Reply by kjmartens** [reply]
This is weird because I cannot duplicate this behavior at all. I have tried a wide variety of approaches and can't make it happen.

Are you able to provide more context about your setup? For example, is this a registration form or an "edit profile" kind of form (where it reloads field data)? And what does your formatting template code look like, etc? 🙂

**Reply by vonscriptor** [reply]
The context is in an edit profile page using a simple render:

```
{{ freeform.form("editProfile").render({
    formAttributes: { "data-skip-html-reload": true },
    submitClass: 'button',
    elementId: currentUser.id
}) }}
```

The form doesn't use AJAX. The field is a Freeform Website type with an Integration to Users, mapped to a Craft CMS URL field type (which seems to be the root of the issue). The Freeform Script Insertion Location is set to none and added using `{{ freeform.loadFreeformPlugin() }}` in my `base.twig`.

Let me know if there's any specific info you'd like to know. For now I've set the Craft CMS field to Plain Text which avoids the issue.

**Reply by kjmartens** [reply]
Ok, I can finally duplicate the issue under this exact scenario:

- Freeform 5
- Craft 4
- User Element integration
- Editing element (`elementId`)
- Leaving the Website field empty.

Upon successful submit, `http://` is automatically inserted it seems.

I've noted this and we'll see what we can do about it. In the meantime, I suppose you stick with your Plain Text field. 🙂
